src/Tuura/Boolean.hs

```diff
@@ -15,9 +17,22 @@ import Tuura.Concept.Circuit
 
 newtype CNF a = CNF { fromCNF :: [[Literal a]] }
 
+instance Show (CNF String) where
+  show = eqShow "*" " + " ((flip (++) ")") . (++) "(") fromCNF
```

Comment:
The implementation stays the same: 
```haskell
pretty :: CNF String -> String
pretty = eqShow "*" " + " ((flip (++) ")") . (++) "(") fromCNF
```

This will not violate the usual expectation that `show` produces valid Haskell (it's not always followed, but it's a useful convention).